xilem_core/src/views/adapt.rs

```diff
@@ -0,0 +1,186 @@
+use core::marker::PhantomData;
+
+use crate::{DynMessage, MessageResult, Mut, View, ViewId, ViewPathTracker};
+
+/// A view that wraps a child view and modifies the state that callbacks have access to.
+///
+/// # Examples
+///
+/// Suppose you have an outer type that looks like
+///
+/// ```ignore
+/// struct State {
```

Comment:
Please don't call this `State`; mixing and matching the name of the generic parameter and its value is a recipe for confusion

Reply:
That resulted from copying the old source, but I agree. I'll rename it to `AppState` to stay consistent. We should probably generally stay consistent with one name for the `AppState` as well (as I've seen `AppData` sometimes).

Reply:
I'd probably prefer these to be idiomatically named something more instructive. Like, maybe `TodoList` or `Tasks` in this case.

For example, in `xilem_minesweeper`, I'd be inclined to call it `Minesweeper` (or indeed `Game`, as it is now)